src/utils.js

```diff
@@ -21,4 +21,54 @@ export function normalizeAxis(axis, computedStyle) {
   }
 
   return axis;
+}
+
+/**
+ * Split an input string into a list of individual component value strings,
+ * so that each can be handled as a keyword or parsed with `CSSNumericValue.parse()`;
+ *
+ * Examples:
+ * splitIntoComponentValues('cover'); // ['cover']
+ * splitIntoComponentValues('auto 0%'); // ['auto', '100%']
+ * splitIntoComponentValues('calc(0% + 50px) calc(100% - 50px)'); // ['calc(0% + 50px)', 'calc(100% - 50px)']
+ * splitIntoComponentValues('1px 2px').map(val => CSSNumericValue.parse(val)) // [new CSSUnitValue(1, 'px'), new CSSUnitValue(2, 'px')]
+ *
+ * @param {string} input
+ * @return {string[]}
+ */
+export function splitIntoComponentValues(input) {
+  const res = [];
+  let i = 0;
+
+  function consumeComponentValue() {
+    let level = 0;
+    const startIndex = i;
+    while (i < input.length) {
+      const char = input.slice(i, i + 1);
+      if (level === 0 && char === ' '){
+        break;
+      } if (char === '(') {
+        level +=1;
+      } else if (char === ')') {
+        level -=1;
+      }
+      i++;
+    }
+    return input.slice(startIndex, i);
+  }
+
+  function consumeWhitespace() {
+    while (input.slice(i, i + 1) === ' ') {
```

Comment:
Instead of checking for a space, we should check for any type of whitespace (`/\s/`).